### PR TITLE
update git remote -v string regex parsing logic

### PR DESCRIPTION
**Pull request by eric-j**
https://new-relic.atlassian.net/browse/NR-276351

### CODE REVIEW

File: shared/agent/src/git/parsers/remoteParser.ts [hunk 1]
```diff
@@ -52,45 +53,52 @@ export class GitRemoteParser {
...
-					path,
+		const lines = data.trim().split("\n");
+
+		for (const line of lines) {
+			const match = remoteRegex.exec(line);
+			if (match) {
+				console.log("Full match:", match[0]);
+				console.log("Remote name:", match[1]);
+				console.log("URL:", match[2]);
+				console.log("Operation:", match[3]);
```
Comment (dsellarsnr) [inline shared/agent/src/git/parsers/remoteParser.ts]: Logger.debug, single log line

File: shared/agent/src/git/parsers/remoteParser.ts [hunk 2]
```diff
@@ -52,45 +53,44 @@ export class GitRemoteParser {
...
-					// Stops excessive memory usage -- https://bugs.chromium.org/p/v8/issues/detail?id=2869
-					` ${match[1]}`.substr(1),
-					url,
-					scheme,
-					domain,
-					path,
+		const lines = data.trim().split("\n");
+
+		for (const line of lines) {
+			const match = remoteRegex.exec(line);
```
Comment (dsellarsnr) [inline shared/agent/src/git/parsers/remoteParser.ts:59]: better be safe and not use exec

File: shared/agent/src/git/parsers/remoteParser.ts [hunk 3]
```diff
@@ -52,45 +53,44 @@ export class GitRemoteParser {
...
-					[{ url: url, type: ` ${match[3]}`.substr(1) as GitRemoteType }]
-				);
-				remotes.push(remote);
-				groups[uniqueness] = remote;
+					remote.types.push({ url: url, type: ` ${match[3]}`.substr(1) as GitRemoteType });
+				}
 			} else {
-				// Stops excessive memory usage -- https://bugs.chromium.org/p/v8/issues/detail?id=2869
-				remote.types.push({ url: url, type: ` ${match[3]}`.substr(1) as GitRemoteType });
+				console.warn("No match found for:", line);
```
Comment (dsellarsnr) [inline shared/agent/src/git/parsers/remoteParser.ts:87]: Logger.warn

File: shared/agent/src/git/gitService.ts [hunk 4]
```diff
@@ -831,8 +831,16 @@ export class GitService implements IGitService, Disposable {
...
+			if (!data) {
+				throw new Error("No data returned from git remote command");
+			}
+			let parsedData = await GitRemoteParser.parse(data, repoPath);
+			if (!parsedData) {
+				throw new Error("Failed to parse remote data");
+			}
+			return parsedData;
+		} catch (error) {
+			console.error(`Failed to get repo remotes for ${repoPath}:`, error);
```
Comment (dsellarsnr) [inline shared/agent/src/git/gitService.ts:843]: Logger.error

**Review by dsellarsnr — APPROVED** [review]

**Review by bcanzanella — APPROVED** [review]
nice work guys!

**Pull request merged**